app/controllers/collections_controller.rb

```diff
@@ -131,12 +127,74 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def create_collection_core(name, metadata, owner)
+    name = sanitize_collection_name(name)
```

Comment:
You are calling `sanitize_collection_name(name)` in `web_create_collection` and again here. Get rid of one of them.